runtime/src/snapshot_utils.rs

```diff
@@ -725,7 +752,10 @@ pub fn get_highest_loadable_bank_snapshot(
         get_highest_full_snapshot_archive_slot(&snapshot_config.full_snapshot_archives_dir)?;
     let full_snapshot_file_slot =
         read_full_snapshot_slot_file(&highest_bank_snapshot.snapshot_dir).ok()?;
-    (full_snapshot_file_slot == highest_full_snapshot_archive_slot).then_some(highest_bank_snapshot)
+    let are_storages_flushed =
+        are_bank_snapshot_storages_flushed(&highest_bank_snapshot.snapshot_dir);
+    (are_storages_flushed && (full_snapshot_file_slot == highest_full_snapshot_archive_slot))
```

Comment:
Does this check mean that if flush is disabled, fastboot will not be able to load and boot from last bank snapshot? 

If Yes, then maybe we are too conservative. I think for 99% of the case, even we don't explicitly flush mmap, the mmap may have already been flushed by OS?

Reply:
I think we can return the snapshot file here and let fastboot to start speculatively, which I think should be fine for most cases.

Reply:
ok. make sense. 
I think we can write the flush complete file when we gracefully shutdown the validator. I am pretty sure that the os will flush all mmap files when the process exit. This way, we will be able to fast boot after gracefully shutdown the validator. 
Of course, this is out of scope of this PR. We can do it in future Prs.